streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -315,66 +367,39 @@ public void testRemoveOldBeforeAddNew() {
-        driver.flushState();
-
-        assertEquals(
-            asList(
-                "1:1 (ts: 0)",
-                "1:12 (ts: 0)",
-                "1:2 (ts: 0)",
-                "1:2 (ts: 0)"),
-            proc.processed);
-    }
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                0L)) {
+            final ConsumerRecordFactory<String, String> recordFactory =
+                new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer(), 0L, 0L);
 
-    @Test
-    public void shouldForwardToCorrectProcessorNodeWhenMultiCacheEvictions() {
-        final String tableOne = "tableOne";
-        final String tableTwo = "tableTwo";
-        final StreamsBuilder builder = new StreamsBuilder();
-        final String reduceTopic = "TestDriver-reducer-store-repartition";
-        final Map<String, Long> reduceResults = new HashMap<>();
+            final MockProcessor<String, String> proc = supplier.theCapturedProcessor();
 
-        final KTable<String, String> one = builder.table(tableOne, consumed);
-        final KTable<Long, String> two = builder.table(tableTwo, Consumed.with(Serdes.Long(), Serdes.String()));
+            driver.pipeInput(recordFactory.create(input, "11", "A"));
 
-        final KTable<String, Long> reduce = two
-            .groupBy(
-                (key, value) -> new KeyValue<>(value, key),
-                Grouped.with(Serdes.String(), Serdes.Long()))
-            .reduce(
-                (value1, value2) -> value1 + value2,
-                (value1, value2) -> value1 - value2,
-                Materialized.as("reducer-store"));
-
-        reduce.toStream().foreach(reduceResults::put);
-
-        one.leftJoin(reduce, (value1, value2) -> value1 + ":" + value2)
-            .mapValues(value -> value);
-
-        driver.setUp(builder, stateDir, 111);
-        driver.process(reduceTopic, "1", new Change<>(1L, null));
-        driver.process("tableOne", "2", "2");
-        // this should trigger eviction on the reducer-store topic
-        driver.process(reduceTopic, "2", new Change<>(2L, null));
-        // this wont as it is the same value
-        driver.process(reduceTopic, "2", new Change<>(2L, null));
-        assertEquals(Long.valueOf(2L), reduceResults.get("2"));
-
-        // this will trigger eviction on the tableOne topic
-        // that in turn will cause an eviction on reducer-topic. It will flush
-        // key 2 as it is the only dirty entry in the cache
-        driver.process("tableOne", "1", "5");
-        assertEquals(Long.valueOf(4L), reduceResults.get("2"));
```

Comment:
This logic isn't possible to test with TTD, so it would have to become an full integration test. But, it seems like it's testing the logic of caching and repartitioning, not the logic of the KTableAggregate, so I'm proposing to remove the test.

Thoughts?

Reply:
SGTM

Reply:
+1